File: sdks/python/src/echo_agent_sdk/observed_isolation_values.py
from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True, slots=True)
class ObservedIsolation:
    value: str

    @classmethod
    def new(cls, value: str) -> ObservedIsolation:
        if not isinstance(value, str):
            raise TypeError("observed isolation must be text")
        trimmed = value.strip()
        return cls(trimmed[:512] if trimmed else "unknown")

    @classmethod
    def default(cls) -> ObservedIsolation:
        return cls("unknown")

    def as_str(self) -> str:
        return self.value

File: sdks/typescript/src/observed_isolation_values.ts
/** Observed Subagent isolation value without owning isolation execution. */
export class ObservedIsolation {
  public readonly value: string;

  private constructor(value: string) {
    this.value = value;
    Object.freeze(this);
  }

  public static new(value: string): ObservedIsolation {
    if (typeof value !== "string") throw new TypeError("observed isolation must be text");
    const trimmed = value.trim();
    return new ObservedIsolation(trimmed ? Array.from(trimmed).slice(0, 512).join("") : "unknown");
  }

  public static default(): ObservedIsolation {
    return new ObservedIsolation("unknown");
  }

  public asStr(): string {
    return this.value;
  }
}
